src/integrations/feed/generators.ts

```diff
@@ -0,0 +1,69 @@
+import { FeedConfigGenerator, FeedResponse, IFeedClient } from './types';
+import { Context } from '../../Context';
+import { CachedFeedClient, FeedClient } from './clients';
+import { ioRedisPool } from '../../redis';
+import {
+  FeedPreferencesConfigGenerator,
+  SimpleFeedConfigGenerator,
+} from './configs';
+
+/**
+ * Utility class for easily generating feeds using provided config and client
+ */
+export class FeedGenerator {
+  private readonly client: IFeedClient;
+  private readonly config: FeedConfigGenerator;
+
+  constructor(client: IFeedClient, config: FeedConfigGenerator) {
+    this.client = client;
+    this.config = config;
+  }
+
+  async generate(
+    ctx: Context,
+    userId: string | undefined,
+    feedId: string | undefined,
+    pageSize: number,
+    offset: number,
+  ): Promise<FeedResponse> {
+    const config = await this.config.generate(
+      ctx,
+      userId,
+      feedId,
+      pageSize,
+      offset,
+    );
+    return this.client.fetchFeed(ctx, feedId, config);
+  }
+}
+
+const client = new FeedClient();
+export const cachedFeedClient = new CachedFeedClient(client, ioRedisPool);
+const opts = {
+  includeBlockedTags: true,
+  includeAllowedTags: true,
+  includeBlockedSources: true,
+  includeSourceMemberships: true,
+};
+
+export const feedGenerators: Record<string, FeedGenerator> = Object.freeze({
```

Comment:
I think it would be better for this to have some type other then `string`, it breaks other typescript features down the line with iterators like reduce and such. So something like `Record<FeedVersion, FeedGenerator>` ..or just `type` with predefined values would work better.

Reply:
Agree! Will refactor

Reply:
Should be `FeedVersion` type now